.github/workflows/contracts.yml

```diff
@@ -150,3 +150,40 @@ jobs:
           token: ${{ secrets.NPM_TOKEN }}
           access: public
           dry-run: true # TODO: to be removed once testing of workflow is done
+
+  migrate-and-publish-celo:
+    needs: [build-and-test, lint]
+    if: |
+      startsWith(github.head_ref, 'releases/alfajores')
+    environment: keep-test
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./solidity
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Migrate contracts
+        env:
+          CONTRACT_OWNER_CELO_ACCOUNT_PRIVATE_KEY: |
+            ${{ secrets.KEEP_TEST_CELO_CONTRACT_OWNER_PRIVATE_KEY }}
+        run: npx truffle migrate --reset --network alfajores
+
+      # TODO: if Celo works with Tenderly, add appropriate push step
+
+      - uses: google-github-actions/setup-gcloud@v0.2.0
+        with:
+          project_id: ${{ secrets.GOOGLE_PROJECT_ID }}
+          service_account_key: ${{ secrets.KEEP_TEST_GCR_JSON_KEY }}
+
+      - name: Upload contract data
+        env:
+          CONTRACT_DATA_BUCKET: ${{ secrets.KEEP_TEST_CONTRACT_DATA_BUCKET }}
+        run: |
+          cd build/contracts
+          gsutil -m cp * gs://"$CONTRACT_DATA_BUCKET"/keep-core-celo
```

Comment:
As we name the directory here `keep-core-celo`, should we rename it for ethereum to `keep-core-ethereum`?

Reply:
I thought about it but decided to leave it in order to maintain backward compatibility. There are several CI jobs that read from that directory and renaming this will require changing all places and moving the files on GCS bucket. I prefer limiting the number of changes in existing jobs and focus on Celo so I think we should leave it as is. After all, it's a quite common assumption that resources without the chain prefix/suffix refer to the default ethereum host chain.

Reply:
Sure thing, I wasn't even thinking about changing it in this PR. Let's just keep it in mind for the future.